<?xml version="1.0"?>
<launch>

    <!-- autonomous navigation move base configuration for turtlebot3 -->

    <!-- Arguments -->
    <arg name="world_name" default="turtlebot3_world"/>
    <arg name="move_forward_only" default="true"/>
    <arg name="cmd_vel_topic" default="/cmd_vel" />
    <arg name="odom_topic" default="odom" />

    <!-- Map server -->
    <node pkg="map_server" name="map_server" type="map_server" args="$(find rdl_turtlebot3_navigation)/maps/$(arg world_name)/map.yaml"/>

    <!-- AMCL -->
    <include file="$(find turtlebot3_navigation)/launch/amcl.launch"/>

    <!-- move_base -->
    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
        <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
        <rosparam file="$(find rdl_turtlebot3_navigation)/config/move_base/costmap_common_params_burger.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find rdl_turtlebot3_navigation)/config/move_base/costmap_common_params_burger.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find rdl_turtlebot3_navigation)/config/move_base/local_costmap_params.yaml" command="load" />
        <rosparam file="$(find rdl_turtlebot3_navigation)/config/move_base/global_costmap_params.yaml" command="load" />
        <rosparam file="$(find rdl_turtlebot3_navigation)/config/move_base/move_base_params.yaml" command="load" />
        <rosparam file="$(find rdl_turtlebot3_navigation)/config/move_base/dwa_local_planner_params_burger.yaml" command="load" />
        <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
        <remap from="odom" to="$(arg odom_topic)"/>
        <param name="DWAPlannerROS/min_vel_x" value="0.0" if="$(arg move_forward_only)" />
    </node>

</launch>
